spec/models/post_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'is valid with a title, with comments_counter and likes_counter greater than or equal to zero' do
+    user = User.create(name: 'Sam', posts_counter: 0)
+    post = Post.new(user: user, title: 'Post for testing', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
+  it 'is invalid without a title' do
+    user = User.create(name: 'Sam', posts_counter: 0)
+    post = Post.new(user: user, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is invalid with a long title (> 250)' do
+    user = User.create(name: 'Sam', posts_counter: 0)
+    post = Post.new(user: user, title: 'a' * 255, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is invalid with a comments_counter less than zero' do
+    user = User.create(name: 'Tom', posts_counter: 0)
+    post = Post.new(user: user, title: 'Test Post', comments_counter: -1, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
+
+  it 'is invalid with a non-integer likes_counter' do
+    user = User.create(name: 'Tom', posts_counter: 0)
+    post = Post.new(user: user, title: 'Post for testing', comments_counter: 0, likes_counter: 1.5)
+    expect(post).to_not be_valid
+  end
+
+  describe '#update_user_post_counter' do
+    it 'increments the user posts_counter when a post is saved' do
+      user = User.create(name: 'Ted', posts_counter: 0)
+      post = Post.new(user: user, title: 'Test Post', comments_counter: 0, likes_counter: 0)
+
+      expect do
+        post.save
+        user.reload
+      end.to change(user, :posts_counter).by(1)
+    end
+  end
+end
```

Comment:
- [ ] Well done for testing the `comments_counter`  and `likes_counter` so far 👏 Kindly make sure to test the following validations as we need add unit tests for all validations as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.) ⛏✨
  -  **_`likes_counter` must be greater than or equal to zero_**
  - **_`comments_counter` must be an integer_**

- [ ] Also, as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), we need to add tests for all model's methods, therefore, kindly make sure to test the `five_most_recent_comments` method present in `post.rb` 💪

Reply:
Hi @KanzaTahreem thank you for the review.
Changes done: added tests for the likes_counter and comments_counter attributes, and for the five_most_recent_comments method.